src/core/or/circuitbuild.c

```diff
@@ -1250,7 +1250,8 @@ circuit_extend(cell_t *cell, circuit_t *circ)
     return -1;
   }
 
-  if (tor_addr_is_internal(&ec.orport_ipv4.addr, 0) &&
+  if ((tor_addr_is_internal(&ec.orport_ipv4.addr, 0) ||
+       tor_addr_is_internal(&ec.orport_ipv6.addr, 0)) &&
```

Comment:
We can't change the extend code until we've done trac #24403.
Please revert this change and the practracker change.